components/selectDevice/Device.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import transferPlayback from "@/lib/device/transferPlayback";
+
+const Device = ({
+  device,
+  currentDeviceId,
+  isPlaying,
+}: {
+  device: SpotifyApi.UserDevice;
+  currentDeviceId: string;
+  isPlaying: boolean;
+}) => {
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Create inline loading UI
+  const isMutating = isFetching || isPending;
```

Comment:
Also, I'm pretty sure that for this situation we shouldn't have `isPending` here — I just tried it and it was WAY more responsive. This is something to explore elsewhere in the app...